python/cudf/cudf/_lib/table.pyx

```diff
@@ -0,0 +1,74 @@
+from cudf._lib.cudf cimport *
+from libcpp.memory cimport unique_ptr, make_unique
+
+cdef class TableView:
+    """
+    A non-owning view into a set of Columns
+    """
+    def __cinit__(self, columns):
+        cdef gdf_column* c_col
+        for col in columns:
+            c_col = column_view_from_column(col)
+            self.c_columns.push_back(c_col)
+        self.ptr = new cudf_table(self.c_columns)
+        self.own_ptr = True
+
+    def __init__(self, columns):
+        pass
+
+    def __dealloc__(self):
+        cdef i
+        if self.own_ptr:
+            for i in range(self.ptr[0].num_columns()):
+                free_column(self.ptr[0].get_column(i))
+            del self.ptr
```

Comment:
Don't you always want to delete this since it's always create at `__cinit__`?

Reply:
Yup

Reply:
Done